tests/api/test_soc_descriptor_gs.cpp

```diff
@@ -0,0 +1,149 @@
+    tt_xy_pair worker_grid_size = soc_desc.worker_grid_size;
+
+    for (size_t x = 0; x < worker_grid_size.x; x++) {
+        for (size_t y = 0; y < worker_grid_size.y; y++) {
+            tt_logical_coords logical_coords = tt_logical_coords(x, y);
+            tt_virtual_coords virtual_coords = soc_desc.logical_to_virtual_coords(logical_coords);
+            tt_physical_coords physical_coords = soc_desc.logical_to_physical_coords(logical_coords);
+            tt_translated_coords translated_coords = soc_desc.logical_to_translated_coords(logical_coords);
+            
+            // Virtual, physical and translated coordinates should be the same.
+            EXPECT_EQ(physical_coords, virtual_coords);
+            EXPECT_EQ(physical_coords, translated_coords);
+        }
+    }
+}
+
+// Test logical to physical coordinate translation.
+// For the full grid of logical coordinates we expect that there are no duplicates of physical coordinates.
+// For the reverse mapping back of physical to logical coordinates we expect that same logical coordinates are returned as from original mapping.
+TEST(SocDescriptor, SocDescriptorGSLogicalPhysicalMapping) {
+
+    tt_SocDescriptor soc_desc = tt_SocDescriptor(test_utils::GetAbsPath("tests/soc_descs/grayskull_10x12.yaml"));
+
+    std::map<tt_logical_coords, tt_physical_coords> logical_to_physical;
+    std::set<tt_physical_coords> physical_coords_set;
+    tt_xy_pair worker_grid_size = soc_desc.worker_grid_size;
+
+    for (size_t x = 0; x < worker_grid_size.x; x++) {
+        for (size_t y = 0; y < worker_grid_size.y; y++) {
+            tt_logical_coords logical_coords = tt_logical_coords(x, y);
+            tt_physical_coords physical_coords = soc_desc.logical_to_physical_coords(logical_coords);
+            logical_to_physical[logical_coords] = physical_coords;
+
+            // Expect that logical to physical translation is 1-1 mapping. No duplicates for physical coordinates.
+            EXPECT_EQ(physical_coords_set.count(physical_coords), 0);
+            physical_coords_set.insert(physical_coords);
+        }
+    }
+
+    EXPECT_EQ(physical_coords_set.size(), worker_grid_size.y * worker_grid_size.x);
+
+    for (auto it : logical_to_physical) {
+        tt_physical_coords physical_coords = it.second;
+        tt_logical_coords logical_coords = soc_desc.physical_to_logical_coords(physical_coords);
+        
+        // Expect that reverse mapping of physical coordinates gives the same logical coordinates
+        // using which we got the physical coordinates.
+        EXPECT_EQ(it.first, logical_coords);
+    }
+}
+
+// Test logical to virtual coordinate translation.
+// For the full grid of logical coordinates we expect that there are no duplicates of virtual coordinates.
+// For the reverse mapping back of virtual to logical coordinates we expect that same logical coordinates are returned as from original mapping.
+TEST(SocDescriptor, SocDescriptorGSLogicalVirtualMapping) {
```

Comment:
There is a lot of common code between these tests.

Do you think it might be clearer to have everything in a single test_soc_descriptor.cpp file?
Then, for example, this test would have a foreach over three different files, but the rest of the code remains almost the same.

For other tests you might need some constant maps maybe (which map from file to expected values), but I still think it could be cleaner. For example first_worker_core_physical_location in  SocDescriptorTopLeftCore and max_num_harvested_x in tests with harvesting... 

You could also make it explicit that grayskull doesn't support harvesting (EXPECT_THROW), supports default harvesting mask 0 only.

Reply:
I thought about this when writing these tests. I don't really have problems with a lot of common code in tests. I think it would be much easier to debug when in the future some tests fail, since one test is going to be only for one arch. I can try to write some common functions which are arch agnostic so I can have as much code as possible in one common place, with that I agree. Also, I think it is cleaner to have separate files because you are actually testing different features (row harvestign, column harvesting) and you can write that in the comments to make the test clearer. Let me know how you feel about this